SimplenoteScreenshots/SnapshotHelper.swift

```diff
@@ -176,7 +180,11 @@ open class Snapshot: NSObject {
                 simulator = regex.stringByReplacingMatches(in: simulator, range: range, withTemplate: "")
 
                 let path = screenshotsDir.appendingPathComponent("\(simulator)-\(name).png")
-                try image.pngData()?.write(to: path, options: .atomic)
+                #if swift(<5.0)
+                    UIImagePNGRepresentation(image)?.write(to: path, options: .atomic)
+                #else
+                    try image.pngData()?.write(to: path, options: .atomic)
+                #endif
```

Comment:
Just curious what is the difference between those two? The parameters and return values description for [UIImagePNGRepresentation](https://developer.apple.com/documentation/uikit/1624096-uiimagepngrepresentation) and [pngData()](https://developer.apple.com/documentation/uikit/uiimage/1624096-pngdata) are the same. Is it about one being a method and the other being a function, hence using `pndData()` after it was introduced really made the code easier to read?

Reply:
This code is automatically generated by Fastlane, so I'm not sure about the design decisions the authors took.

>  hence using `pndData()` after it was introduced really made the code easier to read?

I'd say so. It's also a more "Swifty" way of writing the code. It calls a method on an object instead of using a global free function (which the Swift API guide lines recommend not to use) and it also `throws` giving better error management capabilities.

My first guess was that the `try image.pngData...` call is something only available in Swift 5.0+, but that doesn't seem the case looking at the docs 🤔 

The fact that there is that `#if swift(<5.0)` check, though, definitely makes me think the authors had some reason to prefer `try image.pngData` over `UIImagePNGRepresentation` in code running on a runtime equal to or later than Swift 5.0.

Reply:
Thanks for the extended answer! 😊